src/cache/assignment-cache.ts

```diff
@@ -40,16 +40,20 @@ export abstract class AbstractAssignmentCache<T extends Map<string, string>>
     return cachedValue === getMD5Hash(key.variationKey);
   }
 
-  private get(key: AssignmentCacheKey): string | undefined {
+  get(key: AssignmentCacheKey): string | undefined {
     return this.delegate.get(this.toCacheKeyString(key));
   }
 
   set(key: AssignmentCacheKey): void {
     this.delegate.set(this.toCacheKeyString(key), getMD5Hash(key.variationKey));
   }
 
-  private toCacheKeyString({ subjectKey, flagKey, allocationKey }: AssignmentCacheKey): string {
-    return [`subject:${subjectKey}`, `flag:${flagKey}`, `allocation:${allocationKey}`].join(';');
+  entries(): AssignmentCacheKey[] {
+    return Array.from(this.delegate.entries()).map(([k]) => JSON.parse(k));
+  }
+
+  protected toCacheKeyString({ subjectKey, flagKey, allocationKey }: AssignmentCacheKey): string {
+    return JSON.stringify({ subjectKey, flagKey, allocationKey });
```

Comment:
this should be more robust and allows us to read and write entries to the cache

Reply:
I'm having trouble seeing the advantage of doing this, and now the order of the properties may not be guaranteed in all Javascript environments, either.

Reply:
@aarsilv I thought the use case for this was if we wanted to bootstrap the assignment cache from localStorage, we'd need a way to read all entries, for example when using with `HybridAssignmentCache`. does that make sense?
Also why is the order of properties important? Can you expand on that?

Reply:
See https://github.com/Eppo-exp/js-client-sdk/pull/80/files#diff-0af8e0f3b42473d884541a0cee0aa99e495165411389ef3d9a4627bf524072ae

Reply:
> this should be more robust and allows us to read and write entries to the cache

Is there a bug you found?

> now the order of the properties may not be guaranteed in all Javascript environments

Good thinking - I bet it will be stable within one environment but yes I can see how this introduces some loss of control over something we need to be consistent. 

Speaking of which these values might not be obfuscated.

<img width="913" alt="Screenshot 2024-06-16 at 3 07 07 PM" src="https://github.com/Eppo-exp/js-client-sdk-common/assets/57361/ac68b685-01f3-4353-8e9b-e9349f50d687">

Would this be a compromise to re-implement it with base64 encoding?

```suggestion
    const encode = (str) => btoa(str); // Using Base64 encoding
    return [`subject:${encode(subjectKey)}`, `flag:${encode(flagKey)}`, `allocation:${encode(allocationKey)}`].join(';');
```

Reply:
👆 Oh I like this as it's obfuscated and will certainly be deterministic in all JS environments too!

I think the _way_ we generate a cache key is not coupled with bootstrapping from local storage or whatever, but I may be missing something!

Reply:
Discussed offline, we're switching to a Set-based cache instead.